<launch>
  <arg name="open_rviz" default="true"/>
  <arg name="ns_1"  default="amigo_1"/>
  <arg name="ns_2" default="amigo_2"/>
  
  <!-- Pose Estimation-->
  <arg name="amigo_1_x_pos" default="2.4375"/>
  <arg name="amigo_1_y_pos" default="-0.325"/>
  <arg name="amigo_1_z_pos" default="0.0"/>
  <arg name="amigo_1_yaw"   default="0.0"/>
  
  <arg name="amigo_2_x_pos" default="0.325"/>
  <arg name="amigo_2_y_pos" default="-1.3"/>
  <arg name="amigo_2_z_pos" default="0.0"/>
  <arg name="amigo_2_yaw"   default="-0.0"/>


  <group ns="$(arg ns_1)/map_merge/">
    <param name="init_pose_x"   value="$(arg amigo_1_x_pos)"/>
    <param name="init_pose_y"   value="$(arg amigo_1_y_pos)"/>
    <param name="init_pose_z"   value="$(arg amigo_1_z_pos)"/>
    <param name="init_pose_yaw" value="$(arg amigo_1_yaw)"/>
  </group>

  <group ns="$(arg ns_2)/map_merge">
    <param name="init_pose_x"   value="$(arg amigo_2_x_pos)"/>
    <param name="init_pose_y"   value="$(arg amigo_2_y_pos)"/>
    <param name="init_pose_z"   value="$(arg amigo_2_z_pos)"/>
    <param name="init_pose_yaw" value="$(arg amigo_2_yaw)"/>
  </group>


  <include file="$(find harmonic_map_multirobot_exploration)/launch/amigo_experiment/gmapping_amigo.launch">
    <arg name="namespace" value="$(arg ns_1)"/>
    <arg name="open_rviz" value="false"/>
    <arg name="open_boundary_extraction" value="false"/>
  </include>

  <node pkg="harmonic_map_multirobot_exploration" type="amigo_multi_boundary_extraction.py" name="be_1_node" ns="$(arg ns_1)" output="screen" />

  <include file="$(find harmonic_map_multirobot_exploration)/launch/amigo_experiment/gmapping_amigo.launch">
    <arg name="namespace" value="$(arg ns_2)"/>
    <arg name="open_rviz" value="false"/>
    <arg name="open_boundary_extraction" value="false"/>
  </include>

  <node pkg="harmonic_map_multirobot_exploration" type="amigo_multi_boundary_extraction.py" name="be_2_node" ns="$(arg ns_2)" output="screen" />
  
  
 <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg ns_1)_tf_broadcaster"  args="$(arg amigo_1_x_pos) $(arg amigo_1_y_pos) $(arg amigo_1_z_pos) $(arg amigo_1_yaw) 0 0  /world /$(arg ns_1)/map 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg ns_2)_tf_broadcaster" args="$(arg amigo_2_x_pos) $(arg amigo_2_y_pos) $(arg amigo_2_z_pos) $(arg amigo_2_yaw) 0 0 /world /$(arg ns_2)/map 100"/>
  
  
  <!-- open map merging-->
  <!-- <group ns="map_merge">
    <node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge" output="screen">
      <param name="robot_map_topic" value="map"/>
      <param name="robot_namespace" value="amigo_"/>
      <param name="merged_map_topic" value="map"/>
      <param name="world_frame" value="world"/>
      <param name="known_init_poses" value="false"/>
      <param name="merging_rate" value="0.00000000000000000000005"/>
      <param name="discovery_rate" value="0.5"/>
      <param name="estimation_rate" value="0.5"/>
      <param name="estimation_confidence" value="1.0"/>
    </node>
  </group> -->
  

  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find harmonic_map_multirobot_exploration)/rviz/multi_amigo_slam.rviz"/>
  </group>

</launch>

<!-- 0.66 1.66 0 3.14 0 0
3 0.66 0 3.14 0 0 -->

 <!-- 
<node pkg="tf" type="static_transform_publisher" name="world_to_$(arg ns_1)_tf_broadcaster" args="0 0 0 0 0 0  /world /$(arg ns_1)/map 100"/>
 <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg ns_2)_tf_broadcaster" args="0 0 0 0 0 0 /world /$(arg ns_2)/map 100"/>
   -->